truffleruby/src/main/java/org/truffleruby/core/rubinius/IONodes.java

```diff
@@ -799,7 +801,9 @@ public int accept(DynamicObject io) {
             final int newFd;
 
             try {
-                newFd = ensureSuccessful(nativeSockets().accept(fd, memoryManager().newPointer(address), addressLength));
+                newFd = getContext().getThreadManager().runUntilResult(this,
+                        () -> ensureSuccessful(nativeSockets().accept(fd, memoryManager().newPointer(address), addressLength)),
+                        () -> nativeSockets().shutdown(fd, SHUT_RD));
```

Comment:
Can the user notice the difference?
I would think not except the backtrace if accept is implemented in Ruby.

The question is: after a call with fcntl + non-blocking accept, what do you do?
If a sleep of x ms, then it's potentially wasting that much time before a new connection which might come right after.
The only way to fix that AFAIK is with a `select()` call, or if there was a blocking `accept()` with a timeout (I think not).
Therefore I think just using `select()` + `accept()` when the fd is returned as readable is simpler here.